nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java

```diff
@@ -110,7 +111,9 @@ protected void doRefresh() {
   }
 
   @Override
-  protected void doCommit(TableMetadata base, TableMetadata metadata) {
+  protected void doCommit(BaseMetadata baseMetadata, BaseMetadata newMetadata) {
+    TableMetadata metadata = (TableMetadata) newMetadata;
+    TableMetadata base = (TableMetadata) baseMetadata;
```

Comment:
If we cast to table metadata I think we can't extract the common code for view,

I am expecting `NessieBaseMetastoreTableOperations` extends `BaseMetastoreTableOperations` 
to handle common code and call one interface doCommitImpl() which will be implemented by `NessieTableOperations` and `NessieViewOperations`.

Similarly for other catalogs.

Reply:
I think this should be doable. I am waiting for feedback on parent [PR](https://github.com/apache/iceberg/pull/9682).